uk.co.compucorp.civicrm.hrcomments/api/v3/Comment.php

```diff
@@ -31,7 +39,14 @@ function civicrm_api3_comment_create($params) {
  * @throws API_Exception
  */
 function civicrm_api3_comment_delete($params) {
-  return _civicrm_api3_comment_soft_delete(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  civicrm_api3_verify_mandatory($params, NULL, ['id']);
+
+  if (CRM_HRComments_BAO_Comment::softDelete($params['id'])) {
+    return civicrm_api3_create_success();
+  }
+  else{
+    throw new API_Exception('Comment cannot be deleted because it does not exist');
```

Comment:
The `findByPK` method already throws an exception when it can't find a record with the given ID. So, if inside `Comment::softDelete` you fetch the comment with `$comment = self::findByPk($id)` you won't need to manually throw the exception here